lib/krill.js

```diff
@@ -201,39 +210,84 @@ PredicateStream.prototype.stats = function ()
  * the corresponding C-syntax string).
  */
 
-function RelationalOperator(name, types, evalfunc)
+function RelationalOperator(options, types, evalfunc)
 {
-	this.name = name;
+	mod_assert.object(options, 'options');
+	mod_assert.object(options.names, 'options.names');
+	mod_assert.arrayOfString(types, 'types');
+	mod_assert.func(evalfunc, 'evalfunc');
+
+	var names = options.names;
+
+	this.names = names;
 	this.types = types.slice(0);
 	this.validate = krillPrimValidateRel;
-	this.print = krillPrimPrintRel;
+	this.printCStyle = krillPrimPrintRelCStyle;
+	this.printLDAP = krillPrimPrintRelLDAP;
 	this.eval = evalfunc;
 }
 
-function LogicalOperator(name)
+function LogicalOperator(names)
 {
-	this.name = name;
+	mod_assert.object(names, 'names');
+
+	this.names = names;
 	this.validate = krillPrimValidateLog;
-	this.print = krillPrimPrintLog;
+	this.printCStyle = krillPrimPrintLogCStyle;
+	this.printLDAP = krillPrimPrintLogLDAP;
+}
+
+function buildLdapNotEqualFilter(lhs, rhs) {
+	return ('(!(' + lhs + '=' + rhs + '))');
 }
 
 var krillOps = {
-    'le': new RelationalOperator(
-	'<=', [ 'number' ], function (a, b) { return (a <= b); }),
-    'lt': new RelationalOperator(
-	'<', [ 'number' ], function (a, b) { return (a <  b); }),
-    'ge': new RelationalOperator(
-	'>=', [ 'number' ], function (a, b) { return (a >= b); }),
-    'gt': new RelationalOperator(
-	'>', [ 'number' ], function (a, b) { return (a >  b); }),
-    'eq': new RelationalOperator(
-	'==', [ 'number', 'string', 'boolean' ],
+    'le': new RelationalOperator({
+		names: {
```

Comment:
There are a few undocumented style conventions that I've been using: keys used in objects are quoted (like 'le' is here), and nesting for inline objects like this is a continuation (i.e., 4-space indents).  Would you mind updating this?  Sorry that's not documented or checked, but I'd like to keep the repo consistent.